src/components/Api.js

```diff
@@ -0,0 +1,90 @@
+class Api {
+  constructor(options) {
+    // constructor body
+    this._baseUrl = options.baseUrl;
+    this._headers = options.headers;
+    this._spinner = document.querySelector(".spinner");
+  }
+
+  _sendRequest(root, options) {
```

Comment:
Nice function for fetching and checking responses

Reply:
thanks )